src/libraries/System.Text.Json/Common/JsonSeparatorNamingPolicy.cs

```diff
@@ -16,6 +16,11 @@ internal abstract class JsonSeparatorNamingPolicy : JsonNamingPolicy
 
         public sealed override string ConvertName(string name)
         {
+            if (string.IsNullOrEmpty(name) || !char.IsUpper(name[0]))
```

Comment:
What's the reason for the IsUpper check?  This is also a culture-sensitive uppercase check rather than an invariant one.

Reply:
Seems to be replicating what exists in the camel case naming policy:

https://github.com/dotnet/runtime/blob/7b662a7e264552ea19a51a0135f2441bd5f141aa/src/libraries/System.Text.Json/Common/JsonCamelCaseNamingPolicy.cs#L10-L13

It's an optimisation specific to camel case conversion that shouldn't apply here. Guessing it's contributing to some of the test failures.